src/test/llm/assisted_decoding_test.cpp

```diff
@@ -0,0 +1,429 @@
+#include <pybind11/embed.h>
+#pragma warning(pop)
+
+#include "../../http_rest_api_handler.hpp"
+#include "../../http_status_code.hpp"
+#include "../../json_parser.hpp"
+#include "../../llm/apis/openai_completions.hpp"
+#include "../../llm/language_model/continuous_batching/servable.hpp"
+#include "../../llm/language_model/continuous_batching/llm_executor.hpp"
+#include "../../server.hpp"
+#include "rapidjson/document.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/writer.h"
+#include "../test_http_utils.hpp"
+#include "../test_utils.hpp"
+
+using namespace ovms;
+
+class AssistedDecodingPipelinesHttpTest : public ::testing::Test {
+protected:
+    static std::unique_ptr<std::thread> t;
+
+public:
+    std::unique_ptr<ovms::HttpRestApiHandler> handler;
+
+    std::vector<std::pair<std::string, std::string>> headers;
+    ovms::HttpRequestComponents comp;
+    const std::string endpointChatCompletions = "/v3/chat/completions";
+    const std::string endpointCompletions = "/v3/completions";
+    std::shared_ptr<MockedServerRequestInterface> writer;
+    std::string response;
+    rapidjson::Document parsedResponse;
+    ovms::HttpResponseComponents responseComponents;
+    static std::shared_ptr<ov::genai::ContinuousBatchingPipeline> cbPipe;
+    static std::shared_ptr<LLMExecutorWrapper> llmExecutorWrapper;
+    ov::genai::GenerationConfig config;
+    std::vector<std::string> expectedMessages;
+
+    static void SetUpTestSuite() {
+        std::string port = "9173";
+        ovms::Server& server = ovms::Server::instance();
+        ::SetUpServer(t, server, port, getGenericFullPathForSrcTest("/ovms/src/test/llm/assisted_decoding_config.json").c_str(), 60);
+
+        try {
+            plugin_config_t tokenizerPluginConfig = {};
+            std::string device = "CPU";
+            ov::genai::SchedulerConfig schedulerConfig;
+            schedulerConfig.max_num_batched_tokens = 256;
+            schedulerConfig.cache_size = 1;
+            schedulerConfig.dynamic_split_fuse = true;
+            schedulerConfig.max_num_seqs = 256;
+            plugin_config_t pluginConfig;
+            // Setting precision to f32 fails on SPR hosts - to be investigated??
+            JsonParser::parsePluginConfig("{\"INFERENCE_PRECISION_HINT\":\"f32\"}", pluginConfig);
+            cbPipe = std::make_shared<ov::genai::ContinuousBatchingPipeline>(getGenericFullPathForSrcTest("/ovms/src/test/llm_testing/facebook/opt-125m"), schedulerConfig, device, pluginConfig, tokenizerPluginConfig);
+            llmExecutorWrapper = std::make_shared<LLMExecutorWrapper>(cbPipe);
+        } catch (const std::exception& e) {
+            SPDLOG_ERROR("Error during llm node initialization for models_path exception: {}", e.what());
+        } catch (...) {
+            SPDLOG_ERROR("Error during llm node initialization for models_path");
+        }
+    }
+
+    int generateExpectedText(std::string prompt, bool addSpecialTokens) {
```

Comment:
this is repeated, i have already seen this code in other _test.cpp

Reply:
That's true, I'll see if I can reuse more parts of the code.

Reply:
It has to be repeated. This method is closely tied to the class and uses many of its members. Extracting it would make it's usage less comfortable and readable. Inheritance of the whole class is not viable due to the usage of static members and methods. So reusing this code would require additional effort that was not planned to be in the scope of this PR.